libraries/botbuilder-dialogs-adaptive/src/actions/httpRequest.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { DialogTurnResult, DialogConfiguration, DialogContext, Dialog } from 'botbuilder-dialogs';
+import { ExpressionProperty, ExpressionPropertyValue } from '../expressionProperty';
+import fetch, * as request from "node-fetch";
+import { Activity } from 'botbuilder-core';
+
+export interface HttpRequestConfiguration extends DialogConfiguration {
+
+    method?: HttpMethod;
+
+    valueType?: string;
+
+    url?: string;
+
+    headers?: object;
+
+    body?: object;
+
+    responseType?: ResponsesTypes;
+
+    resultProperty?: string;
+}
+
+export enum ResponsesTypes {
+    /**
+     * No response expected
+     */
+    None,
+
+    /**
+     * Plain JSON response
+     */
+    Json,
+
+    /**
+     * JSON Activity object to send to the user
+     */
+    Activity,
+
+    /**
+     * Json Array of activity objects to send to the user
+     */
+    Activities
+}
+
+export enum HttpMethod {
+    /**
+     * Http GET
+     */
+    GET,
+
+    /**
+     * Http POST
+     */
+    POST,
+
+    /**
+     * Http PATCH
+     */
+    PATCH,
+
+    /**
+     * Http PUT
+     */
+    PUT,
+
+    /**
+     * Http DELETE
+     */
+    DELETE
+}
+
+export class HttpRequest<O extends object = {}> extends Dialog<O> {
+
+    /**
+     * Http Method
+     */
+    public method?: HttpMethod;
+
+    /**
+     * Http Url
+     */
+    public url?: ExpressionProperty<string>;
+
+    /**
+     * Http Headers
+     */
+    public headers?: ExpressionProperty<any>;
+
+    /**
+     * Http Body
+     */
+    public body?: ExpressionProperty<any>;
+
+    /**
+     * The response type of the response
+     */
+    public responseType?: ResponsesTypes;
+
+    /**
+     * Gets or sets the property expression to store the HTTP response in.
+     */
+    public resultProperty?: string;
+
+    constructor();
+    constructor(method: HttpMethod, url: ExpressionPropertyValue<string>, headers: ExpressionPropertyValue<any>,
+        body: ExpressionPropertyValue<any>,
+        responseType: ResponsesTypes, resultProperty: string);
+    constructor(method?: HttpMethod, url?: ExpressionPropertyValue<string>, headers?: ExpressionPropertyValue<any>,
+        body?: ExpressionPropertyValue<any>,
+        responseType?: ResponsesTypes, resultProperty?: string) {
+        super();
+        this.method = method;
+        this.url = new ExpressionProperty(url);
+        this.headers = new ExpressionProperty(headers);
+        this.body = new ExpressionProperty(body);
+        this.responseType = responseType;
+        this.resultProperty = resultProperty;
+    }
+
+    protected onComputeId(): string {
+        return `HttpRequest[${this.method} ${this.url}]`;
+    }
+
+    public configure(config: HttpRequestConfiguration): this {
+        return super.configure(config);
```

Comment:
>return super.configure(config); [](start = 8, length = 31)

The base class doesn't know how to configure the expressions passed in. You'll need to add logic here to explicitly configure any expressions. Look at ifCondition.config() for an example.